debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/ExportStatus.java

```diff
@@ -89,54 +91,71 @@ public void updateMode(ExportMode modeEnum) {
     }
 
     public void flushToDisk() {
+        // TODO: do not create fresh objects every time, just reuse.
         HashMap<String, Object> exportStatusMap = new HashMap<>();
         List<HashMap<String, Object>> tablesInfo = new ArrayList<>();
-        for (Table t : tableExportStatusMap.keySet()) {
+        for (Map.Entry<Table, TableExportStatus> pair : tableExportStatusMap.entrySet()) {
+            Table t = pair.getKey();
+            TableExportStatus tes = pair.getValue();
             HashMap<String, Object> tableInfo = new HashMap<>();
             tableInfo.put("database_name", t.dbName);
             tableInfo.put("schema_name", t.schemaName);
             tableInfo.put("table_name", t.tableName);
-            tableInfo.put("file_name", tableExportStatusMap.get(t).snapshotFilename);
-            tableInfo.put("exported_row_count", tableExportStatusMap.get(t).exportedRowCountSnapshot);
+            tableInfo.put("file_name", tes.snapshotFilename);
+            tableInfo.put("exported_row_count_snapshot", tes.exportedRowCountSnapshot);
+            tableInfo.put("sno", tes.sno);
             tablesInfo.add(tableInfo);
         }
 
         exportStatusMap.put("tables", tablesInfo);
         exportStatusMap.put("mode", mode);
 
         try {
-            ow.writeValue(f, exportStatusMap);
+            // for atomic write, we write to a temp file, and then
+            // rename to destination file path. This prevents readers from reading the file in a corrupted
+            // state (for example, when the complete file has not been written)
+            tempf = new File(getTempFilePath());
+            ow.writeValue(tempf, exportStatusMap);
+            boolean success = tempf.renameTo(f);
```

Comment:
Is it possible to add some retries here? In case of every error or some specific errors which are retryable?

Reply:
Good point!
I'm not aware really, of what could potentially go wrong here. Maybe permission issues? 🤔 
How about we just log an error message and ignore the error? The flushToDisk is called every 2s anyway so there will be retries.

Reply:
This is running in a separate thread, so yes, as it stands, I guess that thread will be killed if we run into this error.

Reply:
We can ignore if there are retryable errors which might go on the next call of `flushToDisk()`
But the issue here is what if there is non-retryable errors like insufficient storage, or permission issues which requires the entire process to error out?